From 7d0e90ed0186ec687c28675cbaeb0e5398554022 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Wed, 31 Mar 2021 22:16:30 +0300
Subject: [PATCH] [SW-36520] module_lkd: fix fit-ready ignore logic

The status flag "is_fit_ready_ignored" needs to be cleared
for any incoming command rather then "FIT READY" command, that
includes (especially) the KMD_MSG_NA command. The flag was not handled
while receiving KMD_MSG_NA (val = 0), as it no longer part of the legacy
relevant code, it became a meta cmd, and should be taken care of at the
meta command level.

Change-Id: I7d9559a19b1089e9e177be28c9cb4b9e2ea0ce25
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 679edcd..b9ca6b1 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -112,6 +112,15 @@
 	if (is_legacy_comms(sts.cmd.val))
 		return module_lkd_check_legacy();
 
+	/*
+	 * Allows handling of any future FIT_RDY
+	 * command. Cause it means that the current command is any
+	 * but FIT_RDY. In this case there is no danger of a repeat
+	 * FIT_RDY command loop. It allows taking care of a FIT_RDY
+	 * at the FIT_RDY switch case and in the module run command
+	 */
+	sts.is_fit_ready_ignored = false;
+
 	return res;
 }
 
